packages/shared-ux/chrome/navigation/src/services.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FC, useContext } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import { NavigationKibanaDependencies, NavigationServices } from '../types';
+
+const Context = React.createContext<NavigationServices | null>(null);
+
+/**
+ * A Context Provider that provides services to the component and its dependencies.
+ */
+export const NavigationProvider: FC<NavigationServices> = ({ children, ...services }) => {
+  return <Context.Provider value={services}>{children}</Context.Provider>;
+};
+
+/**
+ * Kibana-specific Provider that maps dependencies to services.
+ */
+export const NavigationKibanaProvider: FC<NavigationKibanaDependencies> = ({
+  children,
+  ...dependencies
+}) => {
+  const { core } = dependencies;
+  const { http } = core;
+  const { basePath } = http;
+  const { navigateToUrl } = core.application;
+
+  const loadingCount = useObservable(http.getLoadingCount$(), 0);
+
+  const value: NavigationServices = {
+    basePath,
+    loadingCount,
+    navigateToUrl,
+    navIsOpen: true,
```

Comment:
In the next few iterations, we'll refine how the nav is aware of it's open/closed status, and it will use a local storage setting.